core/app/models/spree/promotion_handler/coupon.rb

```diff
@@ -72,8 +84,8 @@ def handle_present_promotion(promotion)
         return promotion_applied if promotion_exists_on_order?(order, promotion)
 
         unless promotion.eligible?(order, promotion_code: promotion_code)
-          set_promotion_eligibility_error_code(promotion)
-          return (error || ineligible_for_this_order)
+          @errors = promotion.eligibility_errors
```

Comment:
If we need to set `@errors` we can probably make it an attr_accessor and use `errors =` ? What do you think?

Reply:
This will make it writable from outside the class, I tend to keep setters private even to avoid confusion in the future.